### PR TITLE
Docs restructure

### PR DESCRIPTION
**Pull request by ewels**
Results of _[edit: "much more than"]_ a day's hacking by @mashehu and @ewels 

Still a lot left to do, see project board: https://github.com/orgs/nf-core/projects/67/views/1

Deployment preview: https://deploy-preview-2441--nf-core.netlify.app/docs

@netlify /docs

### CODE REVIEW

**Comment by mashehu** [comment]
Sidebar is now working on everything and I tried to introduce a 3rd-level to the usage docs.
We need to decide a place for the API docs, fix the merge conflicts and then we should be good to go for the first roll-out.

File: src/content/docs/tutorials/adding_a_pipeline/first_release.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+special case and the tests should pass, and once they do you can request a review from the
+core team.
+
+What happens next depends on the state of your master branch:
+
+- [ ] If you have developed in such a way that your master branch is clean, .i.e. doesn't have
+      any commits since the initial one, the PR will represent all changes
+      associated with the proposed release, and the core team will use it for review and
+      feedback.
+- [ ] If your master branch already contains changes associated with the release, the core
```
Comment (mirpedrol) [inline src/content/docs/tutorials/adding_a_pipeline/first_release.md:44]: Should those be clickable boxes? On the legend (right side) a blue circle appears, only for this section.

**Review by mirpedrol — COMMENTED** [review]
The new docs and tutorials look very pretty! 

I am half way checking all the pages, but I will leave a couple of comments.
This might be expected on previews, but there are a couple of links which are broken.
- The [link to tools docs](https://github.com/nf-core/website/pull/2441/files#diff-55d88e7ff5da3d939256c45aa304fbc088be7de85c0938c939d21a94b3db4284R96) is broken
- [Link to reference_genomes.md](https://github.com/nf-core/website/pull/2441/files#diff-513535b104d41835ced5e99dc83576c58a1512701d2ec58ea25626cad5c6b6b7R189) is broken, and [here](https://github.com/nf-core/website/pull/2441/files#diff-057f2314de998f545cc0a4dfb8adac05670d2537d65f599f02f48f8169fcc78dR64).

File: src/content/docs/tutorials/adding_a_pipeline/first_release.md (hunk 1 above)
Comment (mirpedrol) [reply src/content/docs/tutorials/adding_a_pipeline/first_release.md:44]: image

File: src/content/docs/tutorials/sync/overview.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+To keep all the nf-core pipelines up-to-date with the latest version of the community standards, we have implemented a synchronisation tool.
+This ensures that updates to the community standards are propagated to all nf-core pipelines.
+
+There are three topics covered in this documentation page:
+
+1. [Merging automated PRs](#merging-automated-prs)
+2. [Manual synchronisation](#manual-synchronisation)
+3. [Setting up a pipeline for syncing retrospectively](#setting-up-a-pipeline-for-syncing-retrospectively)
+4. [Fixing a broken `TEMPLATE` branch](#fixing-a-broken-template-branch)
```
Comment (mirpedrol) [inline src/content/docs/tutorials/sync/overview.md:17]: These sections don't exist on this file, I think those are the right paths

```suggestion
1. [Merging automated PRs](/docs/tutorials/sync/merging_automated_prs)
2. [Manual synchronisation](/docs/tutorials/sync/manual_sync)
3. [Setting up a pipeline for syncing retrospectively](/docs/tutorials/sync/sync_retrospectively)
4. [Fixing a broken `TEMPLATE` branch](/docs/tutorials/sync/fixing_broken_template_branch)
```

**Comment by mirpedrol** [comment]
Last comments :) 
- The link to "join the community" [here](https://github.com/nf-core/website/pull/2441/files#diff-ac34aa1cb6c1c16725e1a79dcc39fd34251266fe0bfdae0a560505f83eff5c43R28) is outdated
- The [link](https://github.com/nf-core/website/pull/2441/files#diff-3d8612fc1a20306146757286246df171d8055c0dd53d00c9c511efa9c9cb6b98R31) to the #test-data Slack channel is pointing to the wrong channel.

**Review by jfy133 — APPROVED** [review]
:cowboy_hat_face:

**Comment by jfy133** [comment]
My only remaining comment currently is that I strongly feel that we need to distinguish between ipeline users docs (of all kinds) vs pipeline developer docs (of all kinds).

I Personally feel splitting the current cards into two sections based on this is a worthwhile cause for that purpose, even if  it makes the page a little londer.

File: src/content/docs/tutorials/sync/overview.md (hunk 2 above)
Comment (ewels) [reply src/content/docs/tutorials/sync/overview.md]: Thanks! I fear that there may be more of these, as we broke up quite a lot of pages and it's easy to leave hanging anchor links like this.

We'll just have to do our best to find and fix them as they crop up.

File: src/content/docs/tutorials/adding_a_pipeline/first_release.md (hunk 1 above)
Comment (ewels) [reply src/content/docs/tutorials/adding_a_pipeline/first_release.md:44]: Agree, these should be regular bullet points I think..

Comment (ewels) [reply src/content/docs/tutorials/adding_a_pipeline/first_release.md]: Updated in bca88c5f

**Comment by ewels** [comment]
These links just go to the PR and not a specific location. Most of them I've been able to find myself, but I don't know where this one was.. Do you remember?

> * The [link to tools docs](https://github.com/nf-core/website/pull/2441/files#diff-55d88e7ff5da3d939256c45aa304fbc088be7de85c0938c939d21a94b3db4284R96) is broken

**Comment by ewels** [comment]
ok, merge conflicts fixed and (I think) all previous URLs still working / redirected in case they are referred to from anywhere.

Still more to do, but I think that this is as good as it's going to get for the merge on this PR.

**Pull request merged**